### ISSUE TITLE
Port password confirmation dialog to vue

### ISSUE THREAD

**Issue by PVince81** [nextcloud/server#33675]
### Steps
1. Check out https://github.com/nextcloud/server/pull/33516 (if not merged yet)
2. Go to users page
3. Create a new user
4. Add a password
5. Click "Add new user"
6. Enter your password in the prompt that appears

### Expected result
Password can be entered.

### Actual result
The password prompt appears but cannot be focussed, because the focus trap is still on the parent dialog, which itself uses Vue.
The password prompt itself is not using Vue, so the focus trap interaction cannot work.

With the update of nextcloud-vue and the introduction of a focus trap for vue-based dialogs, old dialogs like this one cannot work correctly as the focus trap will remain on the old dialog.

To make this work correctly, we need to port the password confirmation dialog to Vue: https://github.com/nextcloud/server/blob/update-nextcloud-vue-6.0.0-beta.2/core/src/OC/password-confirmation.js#L90

We need to do this anyway in general to make the dialog follow accessibility standards.

**Comment by PVince81** [comment]
I've added this to https://github.com/nextcloud/server/issues/32117 to warn that this kind of scenario won't work any more.

**Comment by PVince81** [comment]
original issue was here: https://github.com/nextcloud/server/issues/33637

**Comment by Pytal** [comment]
Since this is in Vue I would say we should port this to https://github.com/nextcloud/nextcloud-vue to benefit from the consolidated effort of our Vue build system

There is also this js-only wrapper lib https://github.com/nextcloud/nextcloud-password-confirmation which we could use as a starting point but would splinter off from the https://github.com/nextcloud/nextcloud-vue build system

What do you think @CarlSchwan @nickvergessen @PVince81?

**Comment by nickvergessen** [comment]
I'm fine with making nextcloud-password-confirmation depend on nextcloud-vue
Sounds like the best way forward if we don't want to have it in server/core directly which would also be fine for me.
